### ISSUE TITLE
Validation failed when using `developer_name` 

### ISSUE THREAD

**Issue by JakobDev** [ximion/appstream#560]
When validating a AppStream file which uses the `developer_name` with AppStream 1.0, it fails with `developer-name-tag-deprecated`. You are even not allowed to overwrite it. When trying to use `--override=developer-name-tag-deprecated=info` I get `Can not override issue tag: It is not allowed to downgrade the severity of tag 'developer-name-tag-deprecated' to one that allows validation to pass.`.

This is a big Problem for 2 reasons:
1. The new developer tag [was introduced on 4. October](https://github.com/ximion/appstream/commit/c3f5db4b51fa4bcd71a4e54ebb27e5f142cd1fe2), which was not even 2 months ago. You should give the developers more time to react.
2. I simply can't use the new tag. I publish my programs on Flathub, which is still using appstream-glib. And there are no plans to change this currently, so I'm forced to create a AppStream file that is compatible with appstream-glib and libappstream.

**Comment by ximion** [comment]
> 2\. I simply can't use the new tag. I publish my programs on Flathub, which is still using appstream-glib. And there are no plans to change this currently,

I am expecting this to happen any minute now: https://github.com/flatpak/flatpak-builder/pull/517#issuecomment-1718901343 - and tbh, appstream-glib has been deprecated since 2020 and I just can't keep maintaining support for some incredibly outdated stuff anymore (even before early 2020, asglib was not keeping up with spec additions).

However, putting that aside, I think this warning should be overridable for sure, so I should add it to the whitelist. For the time being, since this was a late change, this also should not have been a warning yet though - 1.0 removed support for all deprecated stuff that had no or almost no users and turned deprecated stuff with a lot of users into warnings.
For the `developer_name` element, it was an oversight that the hint about its deprecation was incredibly recent (0.16.4 shows it) and wasn't like the other changes.

So, tl;dr, we should give people at least a year, maybe more, to change this before turning it into an (overridable) warning, then at a much later point make it non-overridable and at some other point way into the future, remove the old support (or potentially never remove support...).

**Comment by JakobDev** [comment]
> I am expecting this to happen any minute now

Sorry to destroy your hope, but barthalion is currently working on other things, so I think this will take a few more months.  And Yes, I'm also sick of appstream-glib and I'm already waiting a long time, but it looks like this is seen as much important.

You also need to consider that Flatpak is used on LTS Distros, so the AppStream Catalogue should be readable by older AppStream versions. This applies of course only for the features that are already in this old version.

**Comment by ximion** [comment]
> Sorry to destroy your hope, but barthalion is currently working on other things, so I think this will take a few more months.

The thing that annoys me greatly about this is that there is nothing more to do than flip a switch, as far as I know. There is nothing missing in AppStream, and all code has been written. So, as far as I know, there is no way I could help out to accelerate this.

> You also need to consider that Flatpak is used on LTS Distros, so the AppStream Catalogue should be readable by older AppStream versions. This applies of course only for the features that are already in this old version.

The output of 1.0 should be fully compatible with that of 0.16.4 (minus any new features). After briefly breaking Ubuntu and fixing asglib, the output might even work with appstream-glib. But given that Flatpak itself has been using libappstream for years now, that's likely not even a concern anymore.

**Comment by JakobDev** [comment]
> The thing that annoys me greatly about this is that there is nothing more to do than flip a switch, as far as I know.

Changing [org.flatpak.Builder](https://github.com/flathub/org.flatpak.Builder) to use libappstream should be enough. The linter and the website backend probably need patches for the new `developer` element. Flathub is working on allowing Users to directly upload Flatpaks instead of using the Buildserver. That means developers don't need to use what is included in `org.flatpak.Builder`, so they could use libappstream in theory. but it also means, they could still use appstream-glib.

I [got told yesterday](https://github.com/flathub/flathub/issues/3099#issuecomment-1826891099), that there is still no estimated date for the switch. But there is hope:  Today was posted on the [Matrix channel](https://matrix.to/#/#flathub:matrix.org), that the first Apps start using the new `developer` tag, so the Admins are under pressure to move.

**Comment by hughsie** [comment]
I'm not against merging patches to appstream-glib if that makes migration easier.

**Comment by ximion** [comment]
> I'm not against merging patches to appstream-glib if that makes migration easier.

Thank you! But I think adding more stuff to -glib will actually reduce the pressure to update ;-)
There were a few technical issues back then, but they have all been resolved (in part via patches to asglib!), and now it seems like the only thing we're waiting on is someone to actually flip the switch again.

I've asked a few times if there is anything left to do or anything I can help with, but there doesn't seem to be any remaining item besides testing.
